app/controllers/personal_detail_controller.rb

```diff
@@ -2,6 +2,8 @@
 
 class PersonalDetailController < StepsController
   def edit
+    current_app = current_or_new_snap_application
```

Comment:
What do you think about keeping the controller variable as something like `current_snap_application`, so then it leaves the `ApplicationController` to worry about the "find or create" logic?

So, the controllers just ask for the application and get it, but don't worry about where it comes from?